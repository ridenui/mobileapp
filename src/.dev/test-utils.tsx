// https://testing-library.com/docs/react-native-testing-library/setup
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { light } from '@styles/Themes';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={light}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from '@testing-library/react-native';
export * from 'jest-styled-components';

export { customRender as render };
