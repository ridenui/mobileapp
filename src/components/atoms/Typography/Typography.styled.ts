import styled from 'styled-components/native';
import { WithTheme } from '../../../types/Generic';

const getColor = ({ theme }: WithTheme) => ({
  color: theme.text,
});

export const H1 = styled.Text`
  font-size: 48px;
  font-weight: bold;
  ${getColor};
`;

export const H3 = styled.Text`
  font-size: 24px;
  font-weight: bold;
  ${getColor};
`;

export const Paragraph = styled.Text`
  ${getColor};
`;

export const Overline = styled.Text`
  text-transform: uppercase;
  font-size: 24px;
  font-weight: 200;
  ${getColor};
`;
