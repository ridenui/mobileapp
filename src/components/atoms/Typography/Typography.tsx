import React from 'react';
import * as S from './Typography.styled';

export enum TypographyVariants {
  H1 = 'H1',
  Paragraph = 'Paragraph',
  Overline = 'Overline',
}

export interface TypographyProps {
  /** Typography Content. */
  children: string;
  /** Typography Type */
  variant: TypographyVariants;
}

/**
 * Description of Typography.
 */
export function Typography({
  children,
  variant,
}: TypographyProps): JSX.Element {
  const Tag = S[variant];
  return <Tag>{children}</Tag>;
}
