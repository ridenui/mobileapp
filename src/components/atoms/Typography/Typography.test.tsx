import React from 'react';
import { render } from '@dev/test-utils';
import { Typography, TypographyVariants } from './Typography';

test('Typography', () => {
  const { toJSON } = render(
    <Typography variant={TypographyVariants.Overline}>Typography</Typography>,
  );
  expect(toJSON()).toMatchSnapshot();
});
