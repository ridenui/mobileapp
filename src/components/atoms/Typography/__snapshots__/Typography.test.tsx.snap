// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 24,
        "fontWeight": "200",
        "textTransform": "uppercase",
      },
    ]
  }
>
  Typography
</Text>
`;
